client/components/repeater.js

```diff
@@ -88,7 +88,7 @@ class Repeater extends Component {
     return Promise.resolve()
       .then(this.props.onBeforeRemove)
       .then(() => {
-        if (this.props.softDelete) {
+        if (this.props.softDelete && !this.state.items[index].reusable) {
```

Comment:
I don't think this is correct - what is the intent here? 

Note that whilst the input name is `reusable` this is transformed before the step is added to the protocol. To determine if an already created step is reusable the test would be `this.state.items[index].reusableStepId != null`

Reply:
suggested change will not work, as it does not differentiate between an instance of reusable versus reusable object itself. I added following comment in code so that intension of change is clear.
//this is to ensure that reusable step itself is not deleted, as reusable steps may have been used in other places.

Reply:
This is needed, when I create reusable step and then reused in another steps. If I delete reused steps, it also delete the initial step from protocol (so 2 steps are deleted). With this additional check it works as expected.